packages/astro/src/default/components/Logo.astro

```diff
@@ -0,0 +1,43 @@
+---
+import fs from 'node:fs';
+import path from 'node:path';
+
+const LOGO_EXTENSIONS = ['svg', 'png', 'jpeg', 'jpg'];
+
+interface Props {
+  logoLink: string;
+}
+
+function readLogoFile(logoPrefix: string) {
+  let logo;
+
+  for (const logoExt of LOGO_EXTENSIONS) {
+    const logoFilename = `${logoPrefix}.${logoExt}`;
+    const exists = fs.existsSync(path.join('public', logoFilename));
+
+    if (exists) {
+      logo = `/${logoFilename}`;
+      break;
+    }
+  }
+
+  return logo;
+}
+
+const { logoLink } = Astro.props;
+
+const logo = readLogoFile('logo');
+const logoDark = readLogoFile('logo-dark') ?? logo;
+
+if (!logo) {
+  console.warn(`No logo found in public/. Supported filenames are: logo.(${LOGO_EXTENSIONS.join('|')})`);
+}
+---
+
+<a
+  href={logoLink}
+  class="flex items-center text-tk-elements-topBar-logo-color hover:text-tk-elements-topBar-logo-colorHover"
+>
+  {logo && <img class="h-5 w-auto dark:hidden" src={logo} />}
```

Comment:
Should `logo` also fallback to `logo-dark`? I'm just imagining if someone removed `logo.svg` but still has `logo-dark.svg`? Although, maybe that doesn't make sense because it's explicitly `logo-dark`...

Reply:
We discussed this and agreed to keep the current logic where we don't default to `logo-dark` if no `logo` is present.